### PR TITLE
backupccl: report row and byte size in SHOW BACKUP for tenant backups

### LINKED ISSUE #77705: backupccl: SHOW BACKUP does not return row count or file size for tenants
Running SHOW BACKUPS, with multi-tenancy defaulted to on, suggests the backup has no rows:
```
  database_name | parent_schema_name |        object_name         | object_type | backup_type | start_time |          end_time          | size_bytes | rows | is_full_cluster
----------------+--------------------+----------------------------+-------------+-------------+------------+----------------------------+------------+------+------------------
  NULL          | NULL               | movr                       | database    | full        | NULL       | 2022-03-11 21:41:58.661782 |       NULL | NULL |      false
  movr          | NULL               | public                     | schema      | full        | NULL       | 2022-03-11 21:41:58.661782 |       NULL | NULL |      false
  movr          | public             | users                      | table       | full        | NULL       | 2022-03-11 21:41:58.661782 |          0 |    0 |      false
  movr          | public             | vehicles                   | table       | full        | NULL       | 2022-03-11 21:41:58.661782 |          0 |    0 |      false
  movr          | public             | rides                      | table       | full        | NULL       | 2022-03-11 21:41:58.661782 |          0 |    0 |      false
  movr          | public             | vehicle_location_histories | table       | full        | NULL       | 2022-03-11 21:41:58.661782 |          0 |    0 |      false
  movr          | public             | promo_codes                | table       | full        | NULL       | 2022-03-11 21:41:58.661782 |          0 |    0 |      false
  movr          | public             | user_promo_codes           | table       | full        | NULL       | 2022-03-11 21:41:58.661782 |          0 |    0 |      false
(8 rows)

```

This occurs because in `backupShowerDefault()` in `backupccl/show.go`, `encoding.DecodeUvarintAscending` returns an error  (`ERROR: invalid uvarint length of 9`) that we skip handling, and consequently, populating the `descSizes` object.  The backup is still restorable.

Jira issue: CRDB-13729

### PR DESCRIPTION
**Pull request by msbutler**
Previously, if a tenant or system tenant called SHOW BACKUP on a backup created
by a tenant, SHOW BACKUP would return no data on row or byte size. This patch
fixes this.

Fixes #77705

Release justification: low risk bug fix
Release note (sql change): SHOW BACKUP now reports accurate row and byte size
counts on backups created by a tenant.

### CODE REVIEW

**Review by stevendanna — APPROVED** [review]
Thanks for jumping on this.  I've left some nitpicks but nothing blocking.

File: pkg/ccl/backupccl/show.go [hunk 1]
```diff
@@ -541,13 +540,25 @@ func backupShowerDefault(
...
 				}
 				descSizes := make(map[descpb.ID]roachpb.RowCount)
+				var showCodec keys.SQLCodec
+				if len(manifest.Files) > 0 {
+					_, tenantID, err := keys.DecodeTenantPrefix(manifest.Files[0].Span.Key)
+					if err != nil {
+						return nil, err
+					}
+					showCodec = keys.MakeSQLCodec(tenantID)
+				}
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/backupccl/show.go]: good idea! i went even further and put all the row/bytes gathering code in a helper funciton.

File: pkg/ccl/backupccl/show_test.go [hunk 2]
```diff
@@ -555,6 +555,47 @@ func TestShowNonDefaultBackups(t *testing.T) {
...
+	require.Equal(t, systemTenantShowRes, tenant10.QueryStr(t, showBackupQuery, tenantAddr))
+
+	//If the system tenant created the same data, and conducted the same backup,
+	//row counts should look the same
+	systemAddr, httpServerCleanup2 := makeInsecureHTTPServer(t)
+	defer httpServerCleanup2()
+	systemDB.Exec(t, dataQuery)
+	systemDB.Exec(t, backupQuery, systemAddr)
+	require.Equal(t, systemTenantShowRes, systemDB.QueryStr(t, showBackupQuery, systemAddr))
+}
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/backupccl/show_test.go:603]: Done.

[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/backupccl/show_test.go:603]: er, where do you think i should have a newline :D

File: pkg/ccl/backupccl/show_test.go [hunk 3]
```diff
@@ -555,6 +555,47 @@ func TestShowNonDefaultBackups(t *testing.T) {
...
+	showBackupQuery := "SELECT object_name, object_type, rows FROM [SHOW BACKUP FROM LATEST IN $1]"
+	tenant10.Exec(t, dataQuery)
+
+	//First, assert that SHOW BACKUPS on a tenant backup returns the same results if
+	//either the system tenant or tenant10 calls it.
+	tenantAddr, httpServerCleanup := makeInsecureHTTPServer(t)
+	defer httpServerCleanup()
+	tenant10.Exec(t, backupQuery, tenantAddr)
+	systemTenantShowRes := systemDB.QueryStr(t, showBackupQuery, tenantAddr)
+	require.Equal(t, systemTenantShowRes, tenant10.QueryStr(t, showBackupQuery, tenantAddr))
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/ccl/backupccl/show_test.go:593]: yup, ShowBackup tests this.

**Comment by msbutler** [comment]
TFTR!!

bors r=stevendanna

**Pull request merged**